pkg/identity/cache/allocator.go

```diff
@@ -174,7 +174,15 @@ func IdentityAllocationIsLocal(lbls labels.Labels) bool {
 // an identity for the specified set of labels already exist, the identity is
 // re-used and reference counting is performed, otherwise a new identity is
 // allocated via the kvstore.
-func AllocateIdentity(ctx context.Context, lbls labels.Labels) (*identity.Identity, bool, error) {
+func AllocateIdentity(owner IdentityAllocatorOwner, ctx context.Context, lbls labels.Labels) (id *identity.Identity, allocated bool, err error) {
+	defer func() {
```

Comment:
Add a comment indicating what is going on here - this is not obvious.

Reply:
Added:
```
	// Notify the owner of the newly added identities so that the
	// cached identities can be updated ASAP, rather than just
	// relying on the kv-store update events.
```